restructuredtext_lint/cli.py

```diff
@@ -24,37 +27,51 @@ def _main(filepaths, format='text', stream=sys.stdout, encoding=None):
-            elif format == 'json':
-                error_dicts.extend({
-                    'line': error.line,
-                    'source': error.source,
-                    'level': error.level,
-                    'type': error.type,
-                    'message': error.message,
-                    'full_message': error.full_message,
-                } for error in file_errors)
+            file_error_level = max(error.level for error in file_errors)
+            error_level = max(file_error_level, error_level)
+            if file_error_level < report:
+                if format == 'text':
+                    stream.write('INFO File {filepath} has no errors above threshold {report}.\n'.format(
+                            filepath=filepath, report=report))
+            else:
+                if format == 'text':
+                    for error in file_errors:
+                        if error.level >= report:
+                            # e.g. WARNING readme.rst:12 Title underline too short.
+                            stream.write('{err.type} {err.source}:{err.line} {err.message}\n'.format(err=error))
+                elif format == 'json':
+                    error_dicts.extend({
+                        'line': error.line,
+                        'source': error.source,
+                        'level': error.level,
+                        'type': error.type,
+                        'message': error.message,
+                        'full_message': error.full_message,
+                    } for error in file_errors if int(error.level) >= report)
 
     if format == 'json':
         stream.write(json.dumps(error_dicts))
 
-    if error_occurred:
-        sys.exit(1)
+    if error_level >= report:
+        sys.exit(error_level)
     else:
+        # e.g. Run with report level 2, but only found level 1 (info)
         sys.exit(0)
 
 
 def main():
     # Set up options and parse arguments
-    parser = argparse.ArgumentParser(description='Lint reStructuredText files')
+    parser = argparse.ArgumentParser(description="""Lint reStructuredText files.
+
+                                     Returns zero if all files are clean, otherwise returns error
+                                     level 1 to 4 indicating the highest error found with docutils
+                                     (1=info, 2=warning, 3=error, 4=severe).
+                                     """)
     parser.add_argument('--version', action='version', version=VERSION)
     parser.add_argument('filepaths', metavar='filepath', nargs='+', type=str, help='File to lint')
     parser.add_argument('--format', default='text', type=str, help='Format of the output (e.g. text, json)')
     parser.add_argument('--encoding', type=str, help='Encoding of the input file (e.g. utf-8)')
+    parser.add_argument('--report', default=2, type=int, help='Minimum error level to report (1 to 4).')
```

Comment:
Can we add a range here via `choices`? It seems that `argparse` doesn't offer a `max`/`min` so that will work best:

https://docs.python.org/2.7/library/argparse.html#choices

```python
parser.add_argument('--report', default=2, type=int, choices=xrange(1, 4),
  help='Minimum error level to report (1 to 4)')
```